src/StaticAnalyzers.cmake

```diff
@@ -50,6 +50,131 @@ macro(enable_cppcheck CPPCHECK_OPTIONS)
   endif()
 endmacro()
 
+function(_enable_clang_tidy_setup_cl
+  CXX_FLAGS
+  C_FLAGS
+)
+  set(CLANG_TIDY_CXX_FLAGS ${CXX_FLAGS})
+  set(CLANG_TIDY_C_FLAGS ${C_FLAGS})
+
+  if(CMAKE_CXX_STANDARD)
+    list(APPEND CLANG_TIDY_CXX_FLAGS -extra-arg=/std:c++${CMAKE_CXX_STANDARD})
+  endif()
+
+  if(CMAKE_C_STANDARD)
+    list(APPEND CLANG_TIDY_C_FLAGS -extra-arg=/std:c${CMAKE_C_STANDARD})
+  endif()
+
+  set(CLANG_TIDY_CXX_FLAGS ${CLANG_TIDY_CXX_FLAGS} PARENT_SCOPE)
+  set(CLANG_TIDY_C_FLAGS ${CLANG_TIDY_C_FLAGS} PARENT_SCOPE)
+endfunction()
+
+function(_enable_clang_tidy_setup_cross
+  CXX_FLAGS
+  C_FLAGS
+)
+  set(CLANG_TIDY_CXX_FLAGS ${CXX_FLAGS})
+  set(CLANG_TIDY_C_FLAGS ${C_FLAGS})
+
+  # Get GCC default flags
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    execute_process(
+      COMMAND "${CMAKE_CXX_COMPILER}" "-v"
+      ERROR_VARIABLE CLANG_TIDY_CXX_FLAGS_COMPILER_DEFAULT
+      COMMAND_ERROR_IS_FATAL ANY
+    )
+    execute_process(
+      COMMAND "${CMAKE_CXX_COMPILER}" "-dumpmachine"
+      OUTPUT_VARIABLE CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET
+      COMMAND_ERROR_IS_FATAL ANY
+    )
+
+    string(STRIP "${CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET}" CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET)
+    set(CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET "--target=${CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET}")
```

Comment:
There is a DetectCompiler module. I am not sure if this part can be simplified using that. In any case, we can probably refactor it into that module can introduce some macros for compiler/target detection.